x-pack/plugins/reporting/public/components/reporting_panel_content.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// TODO: Remove once typescript definitions are in EUI
+declare module '@elastic/eui' {
+  export const EuiCopy: React.SFC<any>;
+  export const EuiForm: React.SFC<any>;
+}
+
+import { EuiButton, EuiCopy, EuiForm, EuiFormRow, EuiText } from '@elastic/eui';
+import React, { Component } from 'react';
+import { KFetchError } from 'ui/kfetch/kfetch_error';
+import { toastNotifications } from 'ui/notify';
+import url from 'url';
+import { reportingClient } from '../lib/reporting_client';
+
+interface Props {
+  reportType: string;
+  objectId?: string;
+  objectType: string;
+  getJobParams: () => any;
+  options?: any;
+  isDirty: boolean;
+  onClose: () => void;
+}
+
+interface State {
+  isStale: boolean;
+  absoluteUrl?: string;
+}
+
+export class ReportingPanelContent extends Component<Props, State> {
+  private mounted?: boolean;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isStale: false,
+    };
+  }
+
+  public componentWillUnmount() {
+    window.removeEventListener('hashchange', this.markAsStale);
+    window.removeEventListener('resize', this.setAbsoluteReportGenerationUrl);
+
+    this.mounted = false;
+  }
+
+  public componentDidMount() {
+    this.mounted = true;
+    this.setAbsoluteReportGenerationUrl();
+
+    window.addEventListener('hashchange', this.markAsStale, false);
+    window.addEventListener('resize', this.setAbsoluteReportGenerationUrl);
+  }
+
+  public render() {
+    if (this.isNotSaved() || this.props.isDirty || this.state.isStale) {
+      return (
+        <EuiForm className="sharePanelContent" data-test-subj="shareReportingForm">
+          {this.renderGenerateReportButton(true)}
+        </EuiForm>
+      );
+    }
+
+    const reportMsg = `${this.prettyPrintReportingType()}s can take a minute or two to generate based upon the size of your ${
+      this.props.objectType
+    }.`;
+
+    return (
+      <EuiForm className="sharePanelContent" data-test-subj="shareReportingForm">
+        <EuiFormRow>
+          <EuiText>
+            <p>{reportMsg}</p>
+          </EuiText>
+        </EuiFormRow>
+
+        {this.props.options}
+
+        {this.renderGenerateReportButton(false)}
+
+        <EuiFormRow>
```

Comment:
There's a bit more extra functionality that comes with using `EuiFormRow` than just spacing, so if you're not using a form element inside of it, I would stay away from using it as a wrapper and just use `EuiSpacer`s between you're elements.

Reply:
I was using the `EuiFormRow` around the buttons to make them span the entire popup width. Looks like the property that is doing this is `display: flex`. Should I dump `EuiFormRow` and just add the property `display:flex` to the button?

Reply:
You don't need flex, you can just do `style={{ width: `100%` }}`